lib/Controller/SMimeCertificatesController.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @copyright Copyright (c) 2022 Richard Steinmetz <richard@steinmetz.cloud>
+ *
+ * @author Richard Steinmetz <richard@steinmetz.cloud>
+ *
+ * @license AGPL-3.0-or-later
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\Mail\Controller;
+
+use OCA\Mail\Db\SMimeCertificate;
+use OCA\Mail\Db\SMimeCertificateMapper;
+use OCA\Mail\Exception\ServiceException;
+use OCA\Mail\Service\Attachment\UploadedFile;
+use OCA\Mail\Service\SMimeService;
+use OCP\AppFramework\Controller;
+use OCP\AppFramework\Db\DoesNotExistException;
+use OCP\AppFramework\Http\JSONResponse;
+use OCP\IRequest;
+
+class SMimeCertificatesController extends Controller {
+	private ?string $userId;
+	private SMimeService $certificateService;
+	private SMimeCertificateMapper $certificateMapper;
+
+	public function __construct(string $appName,
+								IRequest $request,
+								?string $userId,
+								SMimeService $certificateService,
+								SMimeCertificateMapper $certificateMapper) {
```

Comment:
🚩 mapper in a controller

go through the service

Reply:
Done